internal/archiver/archiver.go

```diff
@@ -216,7 +216,7 @@ func (arch *Archiver) SaveDir(ctx context.Context, snPath string, fi os.FileInfo
 		return FutureTree{}, err
 	}
 
-	names, err := readdirnames(arch.FS, dir)
+	names, err := readdirnames(arch.FS, dir, true)
```

Comment:
just by looking at `readdirnames` calls, I can't tell the meaning of the third parameter. It probably better to split this up into `readdirnames` and `readdirnamessorted` (wrapping the first) functions.

Reply:
In that case, I'd rather just sort on the spot.